index.js

```diff
@@ -202,9 +187,28 @@ Choo.prototype.toString = function (location, state) {
   assert.equal(typeof location, 'string', 'choo.toString: location should be type string')
   assert.equal(typeof this.state, 'object', 'choo.toString: state should be type object')
 
-  this.state.href = location.replace(/\?.+$/, '')
-  this.state.query = nanoquery(location)
-  var html = this.router(location)
+  var href = location.replace(/\?.+$/, '')
+  this._matchRoute(href, location)
+  var html = this._prerender(this.state)
   assert.ok(html, 'choo.toString: no valid value returned for the route ' + location)
   return html.toString()
 }
+
+Choo.prototype._matchRoute = function (location, queryString) {
+  if (location === undefined) location = this._createLocation()
+  if (queryString === undefined) queryString = window.location.search
```

Comment:
note to myself: I shoud add a comment about that `queryString` is only used in the browser.
Or even move it up to the function call... depending how we'd like to handle it in https://github.com/choojs/choo/pull/609